ompi/mpi/fortran/mpif-h/dist_graph_create_f.c

```diff
@@ -72,7 +73,8 @@ void ompi_dist_graph_create_f(MPI_Fint *comm_old, MPI_Fint *n, MPI_Fint *sources
                               MPI_Fint *ierr)
 {
     MPI_Comm c_comm_old, c_comm_graph;
-    int count = 0, i;
+    int __opal_attribute_unused__ count = 0;
```

Comment:
I agree: `count` is *sometimes* used, depending on the size of Fortran integers.

I checked the attribute definition, and it tells the compiler that the variable *may* be unused.  It doesn't mean that the variable *is* unused, so it is safe to use in exactly this situation (i.e., where it *may* or *may not* be used).

FWIW: This PR is a *simple* solution; we could do something more elaborate if you think it is necessary.  I opted for the "attribute indicating it may be unused" solution rather than something like what is shown below because I did not want to replicate the logic of the Fortran integer-size macros in this `.c` file (and some others).

```c
#if SOME_CONDITION_INDICATING_THAT_WE_NEED_THIS_VARIABLE_IN_THE_MACROS_BELOW
   int count = 0;
#endif
```

Reply:
OK, I see that the unused attribute make no changes to the generated code as it only affects the compiler warnings. Enough to satisfy my complaint. I'll mark this as resolved.